lib/broccoli-build-error.js

```diff
@@ -0,0 +1,38 @@
+var path = require('path');
+
+/*
+ * Subclass of `Error`.
+ *
+ * Includes detailed information about the errors from
+ * broccoli plugins, such as error type, code frames,
+ * broccoli node name and package versions.
+ */
+function BroccoliBuildError(originalError, node) {
+  var packageInfo = require(path.join(process.cwd(), 'package.json'));
```

Comment:
`process.cwd()` can be literally anywhere not just in the projects root. I don't believe this approach will work.

I suspect rather, we may need to provide an option to broccoli-builder/middleware, so that ember-cli can provide a callback to "add additional information" to errors on the way out.